packages/flutter/test/material/navigation_bar_test.dart

```diff
@@ -0,0 +1,429 @@
+
+  testWidgets('NavigationBar content test', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      _buildWidget(
+        UniqueKey(),
+        NavigationBar(
+          destinations: const <Widget>[
+            NavigationDestination(
+              icon: Icon(Icons.ac_unit),
+              label: 'AC',
+            ),
+            NavigationDestination(
+              icon: Icon(Icons.access_alarm),
+              label: 'Alarm',
+            ),
+          ],
+          onDestinationSelected: (int i) {},
+        ),
+      ),
+    );
+
+    expect(find.text('AC'), findsOneWidget);
+    expect(find.text('Alarm'), findsOneWidget);
+  });
+
+  testWidgets('NavigationBar can update background color', (WidgetTester tester) async {
+    const Color color = Colors.yellow;
+
+    await tester.pumpWidget(
+      _buildWidget(
+        UniqueKey(),
+        NavigationBar(
+          backgroundColor: color,
+          destinations: const <Widget>[
+            NavigationDestination(
+              icon: Icon(Icons.ac_unit),
+              label: 'AC',
+            ),
+            NavigationDestination(
+              icon: Icon(Icons.access_alarm),
+              label: 'Alarm',
+            ),
+          ],
+          onDestinationSelected: (int i) {},
+        ),
+      ),
+    );
+
+    expect(_getMaterial(tester).color, equals(color));
+  });
+
+  testWidgets('NavigationBar adds bottom padding to height', (WidgetTester tester) async {
+    const double bottomPadding = 40.0;
+
+    await tester.pumpWidget(
+      _buildWidget(
+        UniqueKey(),
+        NavigationBar(
+          destinations: const <Widget>[
+            NavigationDestination(
+              icon: Icon(Icons.ac_unit),
+              label: 'AC',
+            ),
+            NavigationDestination(
+              icon: Icon(Icons.access_alarm),
+              label: 'Alarm',
+            ),
+          ],
+          onDestinationSelected: (int i) {},
+        ),
+      ),
+    );
+
+    final double defaultSize = tester.getSize(find.byType(NavigationBar)).height;
+
+    await tester.pumpWidget(
+      _buildWidget(
+        UniqueKey(),
+        MediaQuery(
+          data: const MediaQueryData(padding: EdgeInsets.only(bottom: bottomPadding)),
+          child: NavigationBar(
+            destinations: const <Widget>[
+              NavigationDestination(
+                icon: Icon(Icons.ac_unit),
+                label: 'AC',
+              ),
+              NavigationDestination(
+                icon: Icon(Icons.access_alarm),
+                label: 'Alarm',
+              ),
+            ],
+            onDestinationSelected: (int i) {},
+          ),
+        ),
+      ),
+    );
+
+    final double expectedHeight = defaultSize + bottomPadding;
+    expect(tester.getSize(find.byType(NavigationBar)).height, expectedHeight);
+  });
+
+  testWidgets('NavigationBar shows tooltips with text scaling ', (WidgetTester tester) async {
+    const String label = 'Foo';
```

Comment:
If this was just one character, it might be easier to explain its expected tooltip size

Reply:
Done